pkg/platform/yaml/yaml.go

```diff
@@ -169,12 +172,14 @@ func (ys *Yaml) WriteTestcase(tc *models.TestCase, ctx context.Context, filters
 
 	if !bypassTestCase {
 		ys.tele.RecordedTestAndMocks()
+		ys.mutex.Lock()
```

Comment:
why is the mutex applied here..?

Reply:
As `testsTotal` is a common variable, if `WriteTestcase` is executed in parallel, it can cause race conditions.